File: 2025-lua/template.lua
local lines = {}
for line in io.lines() do
  table.insert(lines, line)
end

p = 50
pt1 = 0
pt2 = 0
for i, l in ipairs(lines) do
    dir = string.char(string.byte(l, 1))
    amount = tonumber(string.sub(l, 2))

    if dir  == "L" then
        modifier = -1
    end
    if dir == "R" then
        modifier = 1
    end

    p = p + (modifier * amount)
    while p < 0 do
        p = 100 + p
        pt2 = pt2 + 1
    end
    while p > 99 do
        p = 0 + (p - 100)
        pt2 = pt2 + 1
    end
    if p == 0 then
        pt1 = pt1 + 1
    end

end

print("Part 1", pt1)
print("Part 2", pt2)

File: 2025-lua/template.py
import fileinput
import time

time_start = time.time_ns()
lines = [l.strip() for l in fileinput.input()]

pt1 = 0
pt2 = 0

for line in lines:
	pass 

print("--- Day %%DAYNAME%% ---")
print("Part 1", pt1)
print("Part 2", pt2)
print("Took", (time.time_ns() - time_start) // 1_000_000, "ms")

